### ISSUE TITLE
Strange successors of the return block of a function

### ISSUE THREAD

**Issue by DongShuaike** [angr/angr#1861]
I'm analysing a MIPS binary when facing the problem.

The problem exists in the funcition `do_ssc`.

In the following block which has a return statement
![image](https://user-images.githubusercontent.com/8875073/69912053-4fb60900-145f-11ea-9152-4026aab7f033.png)

When I run `node.successors` I got
```
In [103]: end.successors                                                                                              
Out[103]: 
[<CFGNode 0x40a7a8[28]>,
 <CFGNode do_ssc+0x12c [28]>,
 <CFGNode do_ssc+0x4c4 [28]>,
 <CFGNode do_ssc+0x45c [24]>,
 <CFGNode do_ssc+0x2a8 [24]>]
```
Their addresses are `0x40a7a8`, `0x40a33c`, `0x40a6d4` and `0x40a4b8` respectively.

I know the cfg of angr is interfunctional, however, only `0x40a7a8` is an caller of `do_ssc`.

May I know why other threes exist?

**Comment by DongShuaike** [comment]
[ssi.zip](https://github.com/angr/angr/files/3908027/ssi.zip)
The tested binary.

**Comment by ltfish** [comment]
It was because the MIPS ELF fast indirect jump resolver was too aggressive in picking indirect jump targets. Fixed by PR #1862.
